### DISCUSSION TITLE
unicorn-binance-websocket-api 1.41.0

### DISCUSSION THREAD

**Discussion by oliver-zehentleitner** [oliver-zehentleitner/unicorn-binance-websocket-api#268] [Releases]
### Added
- `debug` (bool) parameter to `BinanceWebSocketApiManager()`
- `manager.get_debug_log()`
- `manager.set_socket_is_ready()`
- `manager.set_socket_is_not_ready()`
- `_handle_task_result()` - a callback for eventtasks to retrive exceptions from within the asyncio loop.
### Changed
- `get_new_stream_id()` to `get_new_uuid_id()`
- Improved exception handling and restarts. This should have also positive effects to exection handling within the callback functions.
### Fixing 
- AsyncIO implementation and a temporary workaround for stopping userData streams 
- TRBinance replaced old URL `stream.binance.cc` by the new URL `stream-cloud.trbinance.com` [issue#249](https://github.com/LUCIT-Systems-and-Development/unicorn-binance-websocket-api/issues/249)

Please view the [changelog](https://github.com/LUCIT-Systems-and-Development/unicorn-binance-websocket-api/blob/master/CHANGELOG.md) for further information.

https://pypi.org/project/unicorn-binance-websocket-api/1.41.0
https://anaconda.org/conda-forge/unicorn-binance-websocket-api

<hr /><em>This discussion was created from the release <a href='https://github.com/LUCIT-Systems-and-Development/unicorn-binance-websocket-api/releases/tag/1.41.0'>unicorn-binance-websocket-api</a>.</em>

**Comment by behdad097** [comment]
Why I got below messages one after the other!
BinanceWebSocketApiConnection.await._conn.__aenter__(416449e82a51-6e8f-303b-2182-31120dd6, ['kline_1m'], ['btcusdt']) - ConnectionResetError - error_msg: 
BinanceWebSocketApiConnection.send(416449e82a51-6e8f-303b-2182-31120dd6, ['kline_1m'], ['btcusdt']) - Exception KeyError - error_msg:  '416449e82a51-6e8f-303b-2182-31120dd6'

They print one after the other!!!!

**Reply by oliver-zehentleitner** [reply]
https://github.com/LUCIT-Systems-and-Development/unicorn-binance-websocket-api/issues/new?assignees=oliver-zehentleitner&labels=bug&template=bug_report.yml
